### PR TITLE
chat-js: improve subscription speed by doing two rounds of subscriptions and no longer subscribing to groups

### PR DESCRIPTION
**Pull request by loganallenc**
Groups and permissions have a 1:1 mapping so there's no need to even subscribe to groups.

Perf improvement of about 1-2 seconds on initial pageload!

### CODE REVIEW

**Comment by matildepark** [comment]
T-two seconds is a lot!

File: pkg/interface/chat/src/js/components/lib/chat-input.js [hunk 1]
```diff
@@ -505,29 +476,4 @@ export class ChatInput extends Component {
...
-        if (writePermission.who.has(window.ship)) {
-          return this.writeAccessRender();
-        } else {
-          return this.readOnlyRender();
-        }
-      }
-    } else {
-      return this.writeAccessRender();
-    }
-  }
```
Comment (matildepark) [inline pkg/interface/chat/src/js/components/lib/chat-input.js]: I guess we don’t have read only rooms anymore? Thinking. Thinking. Wait, what if you ban someone? What do they see?

File: pkg/interface/chat/src/js/components/lib/invite-search.js [hunk 2]
```diff
@@ -49,15 +49,17 @@ export class InviteSearch extends Component {
...
+
+    Object.keys(this.props.permissions).map(group => {
+      if (this.props.permissions[group].who.size > 0) {
+        let groupEntries = this.props.permissions[group].who.values();
         for (let member of groupEntries) {
           peerSet.add(member);
         }
       }
+
       if (this.props.contacts[group]) {
```
Comment (matildepark) [inline pkg/interface/chat/src/js/components/lib/invite-search.js]: Wary of letting this component become idiosyncratic. Any way to pass the permissions as a “group” prop?

Comment (loganallenc) [reply pkg/interface/chat/src/js/components/lib/invite-search.js]: Can do that. Will update the PR tomorrow to just pass the set in as “group”

File: pkg/interface/chat/src/js/components/lib/chat-input.js (hunk 1 above)
Comment (loganallenc) [reply pkg/interface/chat/src/js/components/lib/chat-input.js]: We currently show a “you are disconnected from this chatroom” box in that case

Comment (matildepark) [reply pkg/interface/chat/src/js/components/lib/chat-input.js]: Ok then whatev.

**Comment by matildepark** [comment]
I did the thing where I approved too early. lol. You know — awaiting the one change above, this can merge ... will poke when we're ready ...

**Comment by matildepark** [comment]
LGTM, @ixv ready to go

**Comment by loganallenc** [comment]
ready to go

**Pull request merged**